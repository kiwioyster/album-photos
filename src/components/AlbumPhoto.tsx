import React, { useState } from 'react';
import { IPhoto } from '../type';

interface Props {
  photos: IPhoto[];
}

const AlbumPhoto: React.FC<Props> = ({ photos }) => {
  const [shownPhoto, setShownPhoto] = useState<number>();
  const clickThumbnail = (id: number) => {
    setShownPhoto(id);
  };
  return (
    <>
      {photos?.map((photo) => {
        return (
          <div key={photo.id}>
            <img
              className='album-thumbnail'
              src={photo.thumbnailUrl}
              onClick={() => clickThumbnail(photo.id)}
              alt='album thumbnail'
            ></img>
            {(() => {
              if (shownPhoto === photo.id) {
                return (
                  <img
                    className='album-photo-fullsize'
                    alt='album fullsize'
                    src={photos.find((photo) => photo.id === shownPhoto)?.url}
                  ></img>
                );
              }
            })()}
          </div>
        );
      })}
    </>
  );
};

export default AlbumPhoto;
