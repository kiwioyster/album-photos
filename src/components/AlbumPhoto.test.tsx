import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import AlbumPhoto from './AlbumPhoto';
import Adapter from 'enzyme-adapter-react-16';
import { IPhoto } from '../type';
import { cleanup } from '@testing-library/react';

configure({ adapter: new Adapter() });
const mockPhotos: IPhoto[] = [
  {
    albumId: 1,
    id: 2,
    title: 'mock title',
    url: 'mock url',
    thumbnailUrl: 'mock thumbnail url',
  },
  {
    albumId: 2,
    id: 3,
    title: 'mock title 2',
    url: 'mock url 2',
    thumbnailUrl: 'mock thumbnail url 2',
  },
];
describe('AlbumPhoto', () => {
  afterEach(cleanup);
  it('should not show album photo after mount', () => {
    const component = shallow(<AlbumPhoto photos={mockPhotos} />);
    expect(component.find('.album-photo-fullsize').length).toEqual(0);
  });
  it('should show correct thumbnail', () => {
    const component = shallow(<AlbumPhoto photos={mockPhotos} />);
    expect(component.find('.album-thumbnail').at(0).prop('src')).toEqual(
      'mock thumbnail url'
    );
    expect(component.find('.album-thumbnail').at(1).prop('src')).toEqual(
      'mock thumbnail url 2'
    );
  });
  it('should show photo on thumbnail click', () => {
    const component = shallow(<AlbumPhoto photos={mockPhotos} />);
    component.find('.album-thumbnail').at(0).simulate('click');
    expect(component.find('.album-photo-fullsize').length).toEqual(1);
    expect(component.find('.album-photo-fullsize').at(0).prop('src')).toEqual(
      'mock url'
    );
  });
});
