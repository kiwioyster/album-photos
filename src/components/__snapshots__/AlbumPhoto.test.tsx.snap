// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumPhoto should render 1`] = `
Array [
  <React.Fragment>
    <div>
      <img
        alt="album thumbnail"
        className="album-thumbnail"
        onClick={[Function]}
        src="mock thumbnail url"
      />
    </div>
    <div>
      <img
        alt="album thumbnail"
        className="album-thumbnail"
        onClick={[Function]}
        src="mock thumbnail url 2"
      />
    </div>
  </React.Fragment>,
]
`;
